v2/pkgs/tools.go

```diff
@@ -145,31 +148,35 @@ func (c *Tools) Install(ctx context.Context, payload *tools.ToolPayload) (*tools
 		return c.install(ctx, path, *payload.URL, *payload.Checksum)
 	}
 
-	// otherwise we install from the loaded indexes
-	list, err := c.Indexes.List(ctx)
+	// otherwise we install from the default index
+	if !c.Index.IndexFile.Exist() || time.Since(c.Index.LastRefresh) > 1*time.Hour {
+		// Download the file again and save it
+		err := c.Index.DownloadAndVerify()
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	body, err := os.ReadFile(c.Index.IndexFile.String())
```

Comment:
```suggestion
	body, err := t.Index.Read()
```
I'd avoid this code duplication and use a function that encapsulates such logic